### PR TITLE
[Engineering] Move build dependencies feed.

### PR DESCRIPTION
**Pull request by jrbriggs**
Upgrade the feeds to myget enterprise.

### CODE REVIEW

File: nuget.config [hunk 1]
```diff
@@ -5,11 +5,12 @@
   </config>
   <packageSources>
     <clear />
-    <add key="GVFSBuildDependencies" value="https://www.myget.org/F/gvfsbuilddependencies/api/v3/index.json" />
+    <add key="VFSForGitBuildDependencies" value="https://vfsforgit.myget.org/F/build-dependencies/api/v3/index.json" />
+    <add key="ArchivedGVFSBuildDependencies" value="https://vfsforgit.myget.org/F/gvfsbuilddependencies/api/v3/index.json" />
```
Comment (sanoursa) [inline nuget.config]: Why do we need this key? The new feed has all of our latest packages, right? And the older commits still reference the original feed and it's not going away?

File: nuget.config [hunk 2]
```diff
@@ -5,11 +5,12 @@
   </config>
   <packageSources>
     <clear />
-    <add key="GVFSBuildDependencies" value="https://www.myget.org/F/gvfsbuilddependencies/api/v3/index.json" />
+    <add key="VFSForGitBuildDependencies" value="https://vfsforgit.myget.org/F/build-dependencies/api/v3/index.json" />
```
Comment (kewillford) [inline nuget.config]: I'm not seeing this feed in with my user.  Have we pushed any packages to this new feed?  Do we need to update the builds to push to it?

File: nuget.config (hunk 1 above)
Comment (jrbriggs) [reply nuget.config]: `vfsforgit.myget.org/F/build-dependencies` is currently empty.

`vfsforgit.myget.org/F/gvfsbuilddependencies` is what was migrated from `www.myget.org/F/gvfsbuilddependencies`.

I guess another option would be to rename `gvfsbuilddependencies` but I couldn't figure out how to do that in a couple minutes of poking around the myget ui. Given that, I wanted to take this opportunity to move away from the gvfs name.

File: nuget.config (hunk 2 above)
Comment (jrbriggs) [reply nuget.config]: Yes, we'll need to update the relevant builds to push to the new feed.

File: nuget.config (hunk 1 above)
Comment (sanoursa) [reply nuget.config]: I think we should just migrate the latest version of each package over to the new feed, and drop this archived feed from our code altogether. There's not much value in preserving two feeds here.

Comment (jrbriggs) [reply nuget.config]: Makes sense. I'll make those changes.

**Comment by kewillford** [comment]
The API key for myget is also used to push packages for the ProjFS team.  I'm assuming with this new feed there will be a new API key to use and we will need to let them know.

File: nuget.config (hunk 1 above)
Comment (jrbriggs) [reply nuget.config]: I've moved the packages that I think are needed to build this commit. CI should be able to fact check me :)

File: nuget.config (hunk 2 above)
Comment (jrbriggs) [reply nuget.config]: @kewillford do you see the feed when you log into vfsforgit.myget.org?

Comment (jrbriggs) [reply nuget.config]: I've mirrored the permissions from the old feed. Hopefully that fixes the issue.

**Pull request merged**